### PR TITLE
Onprem 5.0.0

### PR DESCRIPTION
**Pull request by bennythejudge**
- start on docs for 5.0.0
- search `TBD` in all docs
- created copying the `4.0.2` directory

### CODE REVIEW

File: 5.0.0/agent_install.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Agent Install
```
Comment (pravin-jothinathan) [inline 5.0.0/agent_install.md:1]: Lets call this as **deprecared**

Comment (pravin-jothinathan) [reply 5.0.0/agent_install.md:1]: We dont want to support agent Installs from Installer anymore.

Comment (bennythejudge) [reply 5.0.0/agent_install.md]: ok done

File: 5.0.0/configuration_parameters.md [hunk 2]
```diff
@@ -0,0 +1,10082 @@
...
+**Required**: `false`<br>
+**Description**: Determines if mysql should run in HA mode.<br>
+**Options**: `true|false`<br>
+**Default**: `false`<br>
+**Example**:
+
+```yaml
+sysdig:
+  mysqlHa: false
+```
```
Comment (jdeiviz) [inline 5.0.0/configuration_parameters.md:1440]: Same here

File: 5.0.0/configuration_parameters.md [hunk 3]
```diff
@@ -0,0 +1,10082 @@
...
+
+```yaml
+sysdig:
+  postgresql:
+    rootPassword: my_root_password
+```
+
+## **sysdig.postgresql.primary**
+**Required**: `false`<br>
+**Description**: If set, the installer starts the mysql to postgresql migration (if not already performed), services will start in postgresql mode.<br>
```
Comment (jdeiviz) [inline 5.0.0/configuration_parameters.md:1821]: Same here

File: 5.0.0/configuration_parameters.md [hunk 4]
```diff
@@ -0,0 +1,10082 @@
...
+**Default**: mysql<br>
+**Example**:
+
+```yaml
+sysdig:
+  secure:
+    scanning:
+      retentionMgr:
+        scanningDBEngine: mysql
+```
```
Comment (jdeiviz) [inline 5.0.0/configuration_parameters.md:5152]: Does this still make sense?

File: 5.0.0/configuration_parameters.md [hunk 5]
```diff
@@ -0,0 +1,10082 @@
...
+**Options**: mysql<br>
+**Default**: "mysql"<br>
+**Example**:
+
+```yaml
+sysdig:
+  secure:
+    veJanitor:
+      scanningDbEngine: "mysql"
+```
```
Comment (jdeiviz) [inline 5.0.0/configuration_parameters.md:9420]: Does this still make sense?

File: 5.0.0/examples/node-labels-and-taints/values.yaml [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+  hostPathNodes:
+    # replace with the name section of kubectl get nodes
+    - my-awesome-node01
+    - my-awesome-node02
+    - my-awesome-node03
+sysdig:
+  mysql:
+    hostPathNodes:
+      # replace with the name section of kubectl get nodes
+      - my-awesome-node01
```
Comment (jdeiviz) [inline 5.0.0/examples/node-labels-and-taints/values.yaml:19]: Same here

File: 5.0.0/examples/openshift-with-hostpath/values.yaml [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+  # Username of the user to access the configured openshift url
+  openshiftUser: <REPLACE_WITH_OPENSHIFT_USERNAME>
+  # Password of the user to access the configured openshift url
+  openshiftPassword: <REPLACE_WITH_OPENSHIFT_PASSWORD>
+  collector:
+    dnsName: <PLEASE_REPLACE>
+  mysql:
+    hostPathNodes:
+      # replace with the name section of kubectl get nodes
+      - my-cool-host1.com
```
Comment (jdeiviz) [inline 5.0.0/examples/openshift-with-hostpath/values.yaml:28]: Same here

File: 5.0.0/examples/single-node/values.yaml [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+# storageClassName: sysdig
+elasticsearch:
+  hostPathNodes:
+      # replace with the name section of kubectl get nodes
+    - minikube
+sysdig:
+  mysql:
+    hostPathNodes:
+      # replace with the name section of kubectl get nodes
+      - minikube
```
Comment (jdeiviz) [inline 5.0.0/examples/single-node/values.yaml:17]: Same here

**Comment by jdeiviz** [comment]
There is a quite a few MySQL occurrences that should be removed, right?

File: 5.0.0/README.md [hunk 9]
```diff
@@ -0,0 +1,364 @@
...
+
+You can perform a quick install if your environment has access to the internet, or a partial or full airgapped installation, as needed. Each is described below.
+
+## About the Installer
+
+The Sysdig Installer tool is a binary containing a collection of scripts that help automate the on-premises deployment of the Sysdig platform (Sysdig Monitor and/or Sysdig Secure), for environments using Kubernetes or OpenShift. Use the Installer to install or upgrade your Sysdig platform. It is recommended as a replacement for the earlier Kubernetes manual installation and upgrade procedures.
+
+## Install vs Upgrade
+
+With Sysdig Platform 3.5.0, the installer has been simplified from previous versions. Upgrade differs from Install in that you run an `installer diff` to discover the differences between the old and new versions and then `installer deploy` for the new version.
```
[the remark this answers is not in the document]
Comment (bennythejudge) [reply 5.0.0/README.md:30]: @jdeiviz do you mean the word `Platform` ? or the entire paragraph?

File: 5.0.0/advanced.md [hunk 10]
```diff
@@ -0,0 +1,401 @@
...
+`storageClassProvisioner`: hostPath.<br>
+`sysdig.cassandra.hostPathNodes`: The number of nodes configured here needs to
+be at minimum 1 when configured `size` is `small`, 3 when configured `size` is
+`medium` and 6 when configured `size` is large.<br>
+`elasticsearch.hostPathNodes`: The number of nodes configured here needs to be
+be at minimum 1 when configured `size` is `small`, 3 when configured `size` is
+`medium` and 6 when configured `size` is large.<br>
+`sysdig.mysql.hostPathNodes`: When sysdig.mysqlHa is configured to true this has
+to be at least 3 nodes and when sysdig.mysqlHa is not configured it should be
+at least one node.<br>
```
[the remark this answers is not in the document]
Comment (bennythejudge) [reply 5.0.0/advanced.md:81]: done

File: 5.0.0/advanced.md [hunk 11]
```diff
@@ -0,0 +1,401 @@
...
+    hostPathNodes:
+      - my-cool-host1.com
+      - my-cool-host2.com
+      - my-cool-host3.com
+      - my-cool-host4.com
+      - my-cool-host5.com
+      - my-cool-host6.com
+  mysql:
+    hostPathNodes:
+      - my-cool-host1.com
```
[the remark this answers is not in the document]
Comment (bennythejudge) [reply 5.0.0/advanced.md:109]: done thanks

File: 5.0.0/configuration_parameters.md [hunk 12]
```diff
@@ -0,0 +1,10082 @@
...
+**Required**: `false`<br>
+**Description**: Specifies the Sysdig Platform components to be installed.<br>
+Combine multiple components by space separating them. Specify at least one
+app, for example, `monitor`.<br>
+**Options**: `monitor|monitor secure|agent|monitor agent|monitor secure agent`<br>
+**Default**: `monitor secure`<br>
+**Example**:
+
+```yaml
+apps: monitor secure
```
[the remark this answers is not in the document]
Comment (bennythejudge) [reply 5.0.0/configuration_parameters.md:67]: done thanks

File: 5.0.0/configuration_parameters.md [hunk 13]
```diff
@@ -0,0 +1,10082 @@
...
+on the host. This is relevant only when `storageClassProvisioner` is
+`hostPath`.<br>
+**Options**:<br>
+**Default**: `/var/lib/mysql`<br>
+**Example**:
+
+```yaml
+hostPathCustomPaths:
+  mysql: `/sysdig/mysql`
+```
```
[the remark this answers is not in the document]
Comment (bennythejudge) [reply 5.0.0/configuration_parameters.md:470]: done thanks

File: 5.0.0/configuration_parameters.md [hunk 14]
```diff
@@ -0,0 +1,10082 @@
...
+[`storageClassProvisioner`](#storageclassprovisioner) is `hostPath`.<br>
+**Options**:<br>
+**Default**: 25Gi<br>
+**Example**:
+
+```yaml
+pvStorageSize:
+  large:
+    mysql: 100Gi
+```
```
[the remark this answers is not in the document]
Comment (bennythejudge) [reply 5.0.0/configuration_parameters.md:559]: Thanks, I have removed all the mysql parameters I could find in this file.

File: 5.0.0/configuration_parameters.md [hunk 15]
```diff
@@ -0,0 +1,10082 @@
...
+**Required**: `false`<br>
+**Description**: The docker image tag for MySQL to PostgreSQL migration.<br>
+**Options**:<br>
+**Default**: 1.2.5-mysql-to-postgres<br>
+**Example**:
+
+```yaml
+sysdig:
+  mysqlToPostgresMigrationVersion: 1.2.5-mysql-to-postgres
+```
```
[the remark this answers is not in the document]
Comment (bennythejudge) [reply 5.0.0/configuration_parameters.md:1778]: thanks for this, I wasn't sure. done

File: 5.0.0/configuration_parameters.md (hunk 3 above)
Comment (bennythejudge) [reply 5.0.0/configuration_parameters.md:1821]: thanks, another one I wasn't sure about

File: 5.0.0/configuration_parameters.md (hunk 4 above)
Comment (bennythejudge) [reply 5.0.0/configuration_parameters.md:5152]: I do not know what this is. Do you?

File: 5.0.0/configuration_parameters.md (hunk 5 above)
Comment (bennythejudge) [reply 5.0.0/configuration_parameters.md:9420]: same as above

Comment (bennythejudge) [reply 5.0.0/configuration_parameters.md:9420]: I need to check the code to find the answer

File: 5.0.0/examples/node-labels-and-taints/values.yaml (hunk 6 above)
Comment (bennythejudge) [reply 5.0.0/examples/node-labels-and-taints/values.yaml:19]: thanks, done

File: 5.0.0/examples/single-node/values.yaml (hunk 8 above)
Comment (bennythejudge) [reply 5.0.0/examples/single-node/values.yaml:17]: thanks, done

File: 5.0.0/configuration_parameters.md (hunk 4 above)
Comment (bennythejudge) [reply 5.0.0/configuration_parameters.md:4743]: https://github.com/draios/installer/blob/onprem-5-0-0/installer/pkg/installer/defaults.go#L428

I have updated accordingly

File: 5.0.0/configuration_parameters.md (hunk 5 above)
Comment (bennythejudge) [reply 5.0.0/configuration_parameters.md:9011]: this is now changed:
https://github.com/draios/installer/blob/onprem-5-0-0/installer/pkg/installer/defaults.go#L428

updating accordingly

File: 5.0.0/examples/README.md [hunk 16]
```diff
@@ -0,0 +1,6 @@
+# Example values.yaml
+
+- [single-node values.yaml](single-node/values.yaml)
+- [openshift-with-hostpath values.yaml](openshift-with-hostpath/values.yaml)
+- [node-labels-and-taints values.yaml](node-labels-and-taints/values.yaml)
+- [elasticsearch-init-vmmaxmapcount_overlays_patch.yaml](elasticsearch-init-vmmaxmapcount/overlays/patch.yaml)
```
[the remark this answers is not in the document]
Comment (pravin-jothinathan) [reply 5.0.0/examples/README.md:6]: I do see that you have added Please add a link here pointing to that README.

File: 5.0.0/release_notes.md [hunk 17]
```diff
@@ -0,0 +1,49 @@
...
+
+Qualified with agent release 11.4.1
+
+### Supported Backend Components
+
+| **Components** | **Kubernetes with Statefulsets** |
+|---|---|
+| Redis                      | 6.0.14 |
+| ElasticSearch              | 6.8.6 |
+| Cassandra                  | Fresh: 3.11.11 - Update: 2.1.22.4 |
```
Comment (pravin-jothinathan) [inline 5.0.0/release_notes.md:33]:
```suggestion
| Cassandra                  | Fresh: 3.11.11 - Upgrade: 2.1.22.4 |
```

**Review by pravin-jothinathan — APPROVED** [review]
thanks @bennythejudge LGTM left couple of small review comments. Can be taken up later after this gets merged.

File: 5.0.0/README.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (jdeiviz) [reply 5.0.0/README.md:30]: I mean the beginning of the sentence:  `With Sysdig Platform 3.5.0, the installer has been simplified from previous versions`. But I'm not all sure thus first confirms with @pravin-jothinathan. This change is not very significant.

**Review by jdeiviz — APPROVED** [review]
Thank you @bennythejudge for taking care of this. Good work!

File: 5.0.0/examples/README.md (hunk 16 above)
[the remark this answers is not in the document]
Comment (bennythejudge) [reply 5.0.0/examples/README.md]: thanks @pravin-jothinathan - I have added it here, pls check again

File: 5.0.0/upgrade_notes.md [hunk 18]
```diff
@@ -0,0 +1,60 @@
...
+
+Fresh installs by default will use Cassandra version 3.x.
+
+### Known limitations for OpenShift 3.11 and Kubernetes < 1.16
+
+PostgreSQL HA is not supported on OpenShift 3.11 and on clusters running Kubernetes < 1.16
+
+### MySql removal (breaking change)
+
+Any MySql related entry in the `values.yaml` must be removed or it will cause the installer to fail.
```
[the remark this answers is not in the document]
Comment (bennythejudge) [reply 5.0.0/upgrade_notes.md:48]: thanks, done

**Review by jdeiviz — APPROVED** [review]

**Review by pablo-calvo — APPROVED** [review]

**Pull request merged**